### ISSUE TITLE
Cannot read an undeclared property Ublaboo\DataGrid\DataSource\DoctrineCollectionDataSource::$data_source.

### ISSUE THREAD

**Issue by JKoty** [contributte/datagrid#652]
I'm trying to use DoctrineCollectionDataSource with custom filter, like this:
```php
$filter = $column->setFilterSelect([/* values */]);
$filter->setCondition(function (Collection $collection, $val) {
	/* filtering */
});
```
But I get this error:
Cannot read an undeclared property **Ublaboo\DataGrid\DataSource\DoctrineCollectionDataSource::$data_source.**
Caused by this:
```php
//FilterableDataSource.php
public function filter(array $filters)
26:        {
27:            foreach ($filters as $filter) {
28:                if ($filter->isValueSet()) {
29:                    if ($filter->hasConditionCallback()) {
30:                        Callback::invokeArgs(
31:                            $filter->getConditionCallback(),
32:                            [$this->data_source /* <= private access*/, $filter->getValue()] /* HERE */
33:                        );
34:                    } else {
```

The problem is obvious - $this->data_source is private. Changing `private` to `protected` solves the problem.
Can we change it?

**Comment by Naxmars** [comment]
Today I had same problem. I created pull request
https://github.com/contributte/datagrid/pull/672

**Comment by f3l1x** [comment]
Resolved via 4aa93a4eb1a38fd84391c96e31d8fe00841f88c1 and #672.
